src/linux/net/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -366,6 +450,19 @@ AccessPointControllerLinux::SetAkmSuites(std::vector<Ieee80211AkmSuite> akmSuite
         return status;
     }
 
+    // Validate that the AKM suites are successfully set by hostapd.
+    auto hostapdBssConfiguration = m_hostapd.GetConfiguration();
+    auto actualKeyManagements = hostapdBssConfiguration.WpaKeyMgmt;
+    if (std::equal(std::begin(wpaKeyManagements), std::end(wpaKeyManagements), std::begin(actualKeyManagements))) {
+        AUDITI << std::format("AKM suites of AP {} set to {}", status.AccessPointId, akms);
+    } else {
+        std::string keyMgmtDebugLog = std::format("AKM suites of AP {} were not set. Current key_mgmt values: ", status.AccessPointId);
+        for (const auto& actualKeyManagement : actualKeyManagements) {
+            keyMgmtDebugLog += std::format("{} ", std::to_underlying(actualKeyManagement));
+        }
```

Comment:
For future reference only, [`std::ostringstream`](https://en.cppreference.com/w/cpp/io/basic_ostringstream) should really be used for this and others building up a string instead as `std::string::operator+=` isn't super-efficient, whereas `std::ostringstream` was made for exactly this scenario. This way, you can also eliminate `std::format` here (and elsewhere) since the stream accepts integer types as well.